[AMORO-2225] Change the default value of `self-optimizing.major.trigger.duplicate-ratio` to 0.1

zhoujinsong opened the pull request:
<!--
Thanks for sending a pull request!

Here are some tips for you:
  1. If this is your first time, please read our contributor guidelines: https://amoro.netease.com/how-to-contribute/
  2. If the PR is related to an issue in https://github.com/NetEase/amoro/issues, add '[AMORO-XXXX]' in your PR title, e.g., '[AMORO-XXXX] Your PR title ...'.
  3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP][AMORO-XXXX] Your PR title ...'.
-->

## Why are the changes needed?
<!--
Please clarify why the changes are needed. For instance,
  1. If you add a feature, you can talk about its use case.
  2. If you fix a bug, you can clarify why it is a bug.
  3. Use Fix/Resolve/Close #{ISSUE_NUMBER} to link this PR to its related issue
-->

Close #225.

## Brief change log
<!--
Clearly describe the changes made in modules, classes, methods, etc.
-->

- Change the default value of `self-optimizing.major.trigger.duplicate-ratio` to 0.1
- Rename property `hive.consistent-write.enabled` to `base.hive.consistent-write.enabled` and move it to HiveTableProperties.
- Polish some codes.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)

baiyangtx reviewed (commented):
There is still code in `TestSparkWriter` that needs to be adjusted and renamed.

zhoujinsong commented:
> There is still code in `TestSparkWriter` that needs to be adjusted and renamed.

Thanks for the reminder, I have fixed the compile error, please take another look.

wangtaohz reviewed (approved):
LGTM
